igvm/commands.py

```diff
@@ -224,32 +228,47 @@ def change_address(vm_hostname, new_address, offline=False):
                     vm.dataset_obj['igvm_operation_mode'])
             )
 
-        old_address = vm.dataset_obj['intern_ip']
         new_address = ip_address(new_address)
-
-        if old_address == new_address:
+        new_network = Query(
+            {
+                'servertype': 'route_network',
+                'state': 'online',
+                'network_type': 'internal',
+                'intern_ip': Contains(new_address),
+            }
+        ).get()['hostname']
+
+        if vm.dataset_obj['intern_ip'] == new_address:
             raise ConfigError('New IP address is the same as the old one!')
 
-        vm_was_running = vm.is_running()
-
-        vm.dataset_obj['intern_ip'] = new_address
-        vm.dataset_obj.commit()
+        if not vm.hypervisor.get_vlan_network(new_address) and not migrate:
+            raise ConfigError('Current hypervisor does not support new subnet!')
 
-        if vm_was_running:
-            vm.shutdown()
+        vm_was_running = vm.is_running()
 
-        try:
-            with Transaction() as transaction:
-                vm.hypervisor.mount_vm_storage(vm, transaction)
+        with Transaction() as transaction:
+            if vm_was_running:
+                vm.shutdown(
+                    transaction=transaction,
+                    check_vm_up_on_transaction=False,
+                )
+            vm.change_address(new_address, new_network, transaction=transaction)
+
+            if migrate:
+                vm_migrate(
+                    vm_object=vm,
+                    run_puppet=True, offline=True, no_shutdown=True,
```

Comment:
same here. shouldn't we check for `offline == True`? e.g. `offline=offline`? what about `offline_transport` then? can we support drbd when `offline=True` and `offline_transport='drbd'`? maybe we need a few more checks here

Reply:
Noppe, this whole function can work only offline and this flag is checked in the beginning of it.